src/odemis/acq/test/flim_test.py

```diff
@@ -272,6 +279,101 @@ def test_setting_stream(self):
         # make sure there are times for each value.
         self.assertEqual(len(self._image.metadata[MD_ACQ_DATE]), self._image.shape[0])
 
+    def test_setting_stream_rep(self):
+        """
+        Check the ROI <> Repetition <> pixelSize connections work fine
+        """
+
+        helper = stream.ScannedTCSettingsStream('Stream', self.apd, self.ex_light, self.lscanner,
+                                                self.sft, self.tc_scanner)
+
+        # Ask full ROI -> should have full ROI
+        helper.roi.value = (0, 0, 1, 1)
+        helper.repetition.value = (256, 256)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0, 0, 1, 1))
+        self.assertEqual(helper.repetition.value, (256, 256))
+
+        pxs256f = helper.pixelSize.value
+
+        # Ask half ROI -> should have half the repetition (and same pixel size)
+        helper.roi.value = (0.5, 0.5, 1, 1)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.5, 0.5, 1, 1))
+        self.assertEqual(helper.repetition.value, (128, 128))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256f)
+
+        # Move the ROI around -> no change in repetition/pixel size
+        helper.roi.value = (0.25, 0.25, 0.75, 0.75)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.25, 0.25, 0.75, 0.75))
+        self.assertEqual(helper.repetition.value, (128, 128))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256f)
+
+        # Increase the ROI by "a little" -> as the laser-mirror doesn't support
```

Comment:
what is a laser-mirror?

Reply:
The laser-mirror is the scanner component in the confocal FLIM.